back/engines/commercial/analytics/app/services/analytics/populate_dimensions_service.rb

```diff
@@ -41,13 +28,9 @@ def populate_types
         ]
 
         current_types = Analytics::DimensionType.all.as_json(only: %i[name parent])
+        return unless current_types.to_set != types.as_json.to_set
 
-        return unless current_types & types != types
```

Comment:
This was actually a bug - this always returned false when both arrays were the same